src/utils/request/request.js

```diff
@@ -0,0 +1,38 @@
+import singleRequest from './singleRequest';
+import applyPlugins from '../applyPlugins';
+import mergeArraysOnProperty from '../mergeArraysOnProperty';
+import fetchAllRequest from './fetchAllRequest';
+
+const request = async (url, fetchOptions = {}, configuration = {}) => {
+  const { plugins: { response: responsePlugins = [] } = {}, fetchAll = false } = configuration;
+
+  if (fetchAll) {
+    const firstRequestData = await singleRequest(url, { ...fetchOptions, page: { number: 1 } });
+
+    if (firstRequestData.meta === undefined) {
+      return applyPlugins(firstRequestData, responsePlugins);
+    }
+
+    const {
+      meta: {
+        page: { size },
+        matches,
+      },
+    } = firstRequestData;
+
+    const amountOfRequests = Math.ceil(matches / size);
```

Comment:
Add a check on matches, because it can be undefined which will result in `amountOfRequests = NaN`.
Maybe we should document it that it only works on responses that contain `matches`.

Reply:
I would do something like

```js
if (firstRequestData.meta === undefined || firstRequestData.meta.matches === undefined) {
  return applyPlugins(firstRequestData, responsePlugins);
}
```

And then additionally add a `console.warn()`?

What do you think?

Reply:
Yes, that should do the trick. 
Not sure about the `console.warn` though. I kind of hate it when packages add console logs. 😄
I certainly don't want it in production.

Reply:
Can we just throw an error? Seems like a better solution as you shoulnd't be using `fetchAll` for endpoints that don't return matching

Reply:
But is it an error?